include/HubFramework/HUBConfigViewControllerFactory.h

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright (c) 2016 Spotify AB.
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+@class HUBConfig;
+@class HUBViewController;
+@protocol HUBActionHandler;
+@protocol HUBContentOperation;
+@protocol HUBViewModel;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ *  `HUBViewController` factory that uses a `HUBConfig` for all the configuration needed to create a view controller.
+ *
+ *  This view controller allows for a fully configured `HUBViewController` to be created without going through the 
+ *  `HUBManager` and `HUBFeatureRegistration`.
+ */
+@interface HUBConfigViewControllerFactory : NSObject
```

Comment:
Initially it didn't make a lot of sense and later I didn't revisit that decision. It could be changed to be configured with a configuration (rather than passing in the configuration to the `create...`-methos). However at least one of the methods in the `HUBViewControllerFactory` protocol does't make sense when used without `HUBManager`/`HUBFeatureRegistry`.